packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -226,9 +226,12 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
             option.label.toLowerCase().includes(value.toLowerCase()),
           ),
       );
+      if (focusedIndex != -1) {
```

Comment:
Why needed?

Reply:
Decided to reset focus whenever user types, so we dont need to handle things like if focused element is visible after typing or not and then do something about `focusedIndex`